runtime/lua/vim/fswatch/fswatch.lua

```diff
@@ -0,0 +1,140 @@
+--[[
+  Create a file watcher, each watcher is identified by the name of the file that it
+  watches. We can use a lua table to store all watchers indexed by their filenames
+  so that we can close the required watcher during the callback to on_change to
+  debounce the watcher.
+--]]
+
+local uv = vim.loop
+
+local Watcher = {
```

Comment:
I don't think you want to put these here like this. You want to define these as defaults for each watcher in `Watcher:new`. Otherwise you might share values like `paused`, which you don't want to do.

Reply:
Oh, I see. I will move the properties local to a watcher in `Watcher:new()` then.